src/components/MobileWalletDialog/routes/CreateWallet/CreateWallet.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  Box,
+  Button,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  Text as CText,
+  VStack,
+} from '@chakra-ui/react'
+import { useCallback, useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import { useHistory, useLocation } from 'react-router'
+import { CarouselDots } from 'components/CarouselDots/CarouselDots'
+import { DialogBackButton } from 'components/Modal/components/DialogBackButton'
+import { DialogBody } from 'components/Modal/components/DialogBody'
+import { DialogCloseButton } from 'components/Modal/components/DialogCloseButton'
+import { DialogFooter } from 'components/Modal/components/DialogFooter'
+import {
+  DialogHeader,
+  DialogHeaderLeft,
+  DialogHeaderMiddle,
+  DialogHeaderRight,
+} from 'components/Modal/components/DialogHeader'
+import { SlideTransition } from 'components/SlideTransition'
+import { Text } from 'components/Text'
+import { createWallet } from 'context/WalletProvider/MobileWallet/mobileMessageHandlers'
+import type { RevocableWallet } from 'context/WalletProvider/MobileWallet/RevocableWallet'
+import type { MobileLocationState } from 'context/WalletProvider/MobileWallet/types'
+
+import { MobileWalletDialogRoutes } from '../../types'
+
+type FormValues = {
+  label: string
+}
+
+export const CreateWallet = () => {
+  const location = useLocation<MobileLocationState | undefined>()
+  const history = useHistory()
+  const translate = useTranslate()
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isSubmitting, isValid },
+  } = useForm<FormValues>({
+    mode: 'onChange',
+    defaultValues: { label: location.state?.vault?.label },
+  })
+  const [vault, setVault] = useState<RevocableWallet | null>(null)
+
+  const onSubmit = useCallback(
+    (values: FormValues) => {
+      if (!vault) return
+
+      try {
+        vault.label = values.label
+        history.push(MobileWalletDialogRoutes.KeepSafe, { vault })
+      } catch (e) {
+        console.log(e)
+      }
+    },
+    [history, vault],
+  )
+
+  const handleBack = useCallback(() => history.push(MobileWalletDialogRoutes.Saved), [history])
+
+  useEffect(() => {
+    try {
+      if (!vault) setVault(location.state?.vault ?? createWallet())
+    } catch (e) {
+      console.log(e)
+    }
```

Comment:
My biggest concern:

Should we revoke the wallet and recreate it in every screens? It's a lot of boilerplate/verbose code that I avoided, because I mean, if a not-memoized component is unmounted, it shouldn't stay in memory

But happy to bring it if we feel like we need some extra safety

Reply:
Think this *should* be safe indeed since this is a state var. 
Revoker basically just revokes whichever secret (seed, master key, isolated mnemonic or whatever the isolation wallet does magic-wise...), but indeed, vault is also nuked from cache on unmount in *theory*.

With that being said,I'm not sure that it being theoretically safe is a good enough reason not to revoke, especially given how routing works (definitely seen many instances of routes still being mounted), and not super confident with security relying on such assumption.

@NeOMakinG can we keep it non-revoked and assume routes being unmounted means garbage collect kicking in and nuking vault for *that* PR but do a direct follow-up with `revoke()` boilerplate?

Reply:
I think. no point to implement it in a follow up considering a potential risk, let's do it here

Reply:
Did it in [226d3bc](https://github.com/shapeshift/web/pull/8760/commits/226d3bc4956d53bc7dbc3339ca20e324640d44c1)